File: data/gpt-4o/readable_format_HPEC/cpp/file245.cpp
void init_array(int n, double a[][n], double x[], double c[]) {
  int i, j;
  for (i = 0; i < n; i++) {
    c[i] = double(i) / double(n);
    x[i] = double(i) / double(n);
    for (j = 0; j < n; j++) {
      a[j][i] = (double(i) * double(j)) / double(n);
    }
  }
}

File: data/StarCoder2/readable_format_HPEC/fortran/file245.f90
subroutine init_array(n, a, x, c)
implicit none

DATA_TYPE, dimension(n, n) :: a
DATA_TYPE, dimension(n) :: c
DATA_TYPE, dimension(n) :: x
integer :: n
integer :: i, j
do i = 1, n
c(i) = DBLE(i - 1) / DBLE(n)
x(i) = DBLE(i - 1) / DBLE(n)
do j = 1, n
a(j, i) = (DBLE(i - 1) * DBLE(j - 1)) / DBLE(n)
end do
end do
end subroutine
